### PR TITLE
Installation instructions in documentation

### PR DESCRIPTION
**Pull request by rrjbca**
## Description
As suggested in #354 move the installation instructions from the README into a new *Installation* page of the documentation to reduce clutter. 

- [Front Page](https://skypy--376.org.readthedocs.build/en/376/)
- [Installation Page](https://skypy--376.org.readthedocs.build/en/376/install.html)

## Checklist
- [x] Follow the [Contributor Guidelines](https://github.com/skypyproject/skypy/blob/master/CONTRIBUTING.md)
- [ ] ~Write unit tests~
- [ ] ~Write documentation strings~
- [ ] ~Assign someone from your working team to review this pull request~
- [x] Assign someone from the infrastructure team to review this pull request

### CODE REVIEW

**Comment by rrjbca** [comment]
After #368 will [skypyproject/skypy-data](https://github.com/skypyproject/skypy-data) continue to be a dependency? If so will it's installation instructions stay the same?

**Comment by ntessore** [comment]
> After #368 will [skypyproject/skypy-data](https://github.com/skypyproject/skypy-data) continue to be a dependency?

I would be in favour of moving the SED templates into SkyPy and removing the dependency (and deprecating the repository). After all, it was a quick fix for a breaking issue when trying to get v0.3 over the line.

**Comment by rrjbca** [comment]
In which case I will suggest not adding any details regarding skypy-data to the documentation and instead we can remove the skypy-data instructions from the README when we deprecate that package.

File: docs/install.rst [hunk 1]
```diff
@@ -0,0 +1,96 @@
+############
+Installation
+############
+
+This page outlines how to install one of the officially distributed SkyPy
+releases and it's dependencies, or install and test the latest development
```
Comment (ntessore) [inline docs/install.rst:6]:
```suggestion
releases and its dependencies, or install and test the latest development
```

File: docs/install.rst [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+.. code:: console
+
+    $ conda install -c conda-forge skypy
+
+From GitHub
+-----------
+
+The latest development version of SkyPy can be found on the master branch of
+the `skypyproject/skypy`_ GitHub repository. This and any other branch or tag
+can be installed directly from GitHub using pip:
```
Comment (ntessore) [inline docs/install.rst:34]:
```suggestion
can be installed directly from GitHub using a recent version ofpip:
```

File: docs/install.rst [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+.. code:: console
+
+    $ pip install git+https://github.com/skypyproject/skypy@master
```
Comment (ntessore) [inline docs/install.rst:38]:
```suggestion
    $ pip install skypy@git+https://github.com/skypyproject/skypy.git@master
```

File: docs/install.rst [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+Installing using pip or conda will automatically install or update these core
+dependencies if necessary. SkyPy also has a number of optional dependencies
+that enable additional features:
+
+- `camb <https://camb.info/>`_
+- `classy <https://lesgourg.github.io/class_public/class.html>`_
+- `healpy <https://healpy.readthedocs.io/>`_
+- `speclite <https://speclite.readthedocs.io/>`_
+- `specutils <https://specutils.readthedocs.io/>`_
+
```
Comment (ntessore) [inline docs/install.rst:61]: Do we want to mention that all of these can be installed with `pip install skypy[all]`?

File: docs/install.rst [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+    $ skypy --version
+
+You may also want to run the unit tests, for example if you have installed the
+development version or you use an unsupported operating system. The tests have
+the following additional dependencies:
+
+- `pytest-astropy <https://github.com/astropy/pytest-astropy>`_
+- `pytest-rerunfailures <https://github.com/pytest-dev/pytest-rerunfailures>`_
+
+and can be run using pytest_:
```
Comment (ntessore) [inline docs/install.rst:85]: All test dependencies can be installed with `pip install skypy[test]`.

**Review by ntessore — APPROVED** [review]

**Pull request merged**